### PR TITLE
Add "do not warn again" checkbox to media preview dialog.

### PR DESCRIPTION
**Pull request by softwareanimal**
This is for bug #1915.

### CODE REVIEW

File: res/values/strings.xml [hunk 1]
```diff
@@ -121,6 +121,7 @@
     <string name="ConversationFragment_are_you_sure_you_want_to_permanently_delete_all_selected_messages">Are you sure that you want to permanently delete all selected messages?</string>
     <string name="ConversationFragment_save_to_sd_card">Save to storage?</string>
     <string name="ConversationFragment_this_media_has_been_stored_in_an_encrypted_database_warning">Saving this media to storage will allow any other apps on your phone to access it.\n\nContinue?</string>
+    <string name="ConversationFragment_do_not_show_this_message_again">Do not show this message again</string>
```
Comment (mcginty) [inline res/values/strings.xml]: Maybe "Don't warn me again" to keep it short and sweet :)

**Comment by softwareanimal** [comment]
Improved the text to just say "Don't warn me again", as suggested.

File: src/org/thoughtcrime/securesms/util/TextSecurePreferences.java [hunk 2]
```diff
@@ -64,11 +64,20 @@
 
   private static final String PUSH_REGISTRATION_REMINDER_PREF  = "pref_push_registration_reminder";
   public  static final String REPEAT_ALERTS_PREF               = "pref_repeat_alerts";
+  public  static final String REPEAT_MEDIA_DECRYPTION_WARNING_PREF = "pref_media_decryption_warning";
```
Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/util/TextSecurePreferences.java]: alignment

File: src/org/thoughtcrime/securesms/util/TextSecurePreferences.java [hunk 3]
```diff
@@ -64,11 +64,20 @@
...
 
   public static boolean isWifiSmsEnabled(Context context) {
     return getBooleanPreference(context, WIFI_SMS_PREF, false);
   }
 
+  public static boolean isMediaDecryptionWarningPreferred(Context context) {
+      return getBooleanPreference(context, REPEAT_MEDIA_DECRYPTION_WARNING_PREF, true);
```
Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/util/TextSecurePreferences.java]: 2 space indent

File: src/org/thoughtcrime/securesms/util/TextSecurePreferences.java [hunk 4]
```diff
@@ -64,11 +64,20 @@
...
+  }
+
+  public static void setMediaDecryptionWarningPreferred(Context context, boolean warn) {
+      setBooleanPreference(context, REPEAT_MEDIA_DECRYPTION_WARNING_PREF, warn);
```
Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/util/TextSecurePreferences.java]: 2 space indent

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 5]
```diff
@@ -509,21 +512,38 @@ public void onClick(View v) {
         if (!messageRecord.isOutgoing()) intent.putExtra(MediaPreviewActivity.RECIPIENT_EXTRA, messageRecord.getIndividualRecipient().getRecipientId());
         intent.putExtra(MediaPreviewActivity.DATE_EXTRA, messageRecord.getDateReceived());
         context.startActivity(intent);
+      } else if ( TextSecurePreferences.isMediaDecryptionWarningPreferred(context) ) {
+          showMediaDecryptionWarning();
```
Comment (agrajaghh) [inline src/org/thoughtcrime/securesms/ConversationItem.java]: 2 space indent

File: src/org/thoughtcrime/securesms/util/TextSecurePreferences.java [hunk 6]
```diff
@@ -64,11 +64,20 @@
 
   private static final String PUSH_REGISTRATION_REMINDER_PREF  = "pref_push_registration_reminder";
   public  static final String REPEAT_ALERTS_PREF               = "pref_repeat_alerts";
+  public  static final String REPEAT_MEDIA_DECRYPT_ALERTS_PREF = "pref_repeat_media_decrypt_alerts";
 
   public static boolean isWifiSmsEnabled(Context context) {
     return getBooleanPreference(context, WIFI_SMS_PREF, false);
   }
 
+  public static boolean isMediaDecryptionWarningPreferred(Context context) {
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/util/TextSecurePreferences.java]: `isMediaPreviewWarningEnabled` may make more sense, as we decrypt to show in memory too.

File: src/org/thoughtcrime/securesms/util/TextSecurePreferences.java [hunk 7]
```diff
@@ -64,11 +64,20 @@
 
   private static final String PUSH_REGISTRATION_REMINDER_PREF  = "pref_push_registration_reminder";
   public  static final String REPEAT_ALERTS_PREF               = "pref_repeat_alerts";
+  public  static final String REPEAT_MEDIA_DECRYPT_ALERTS_PREF = "pref_repeat_media_decrypt_alerts";
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/util/TextSecurePreferences.java]: same for something like `EXTERNAL_MEDIA_WARN_PREF`

**Comment by mcginty** [comment]
On further consideration, I wonder if we should have one "don't warn me again" that applies to both previewing and saving. It could be the same dialog.

"Previewing or saving media outside of TextSecure allows other apps to see the media."

and then we could have "Continue" and "Cancel" buttons, along with a "Don't warn me again".

**Comment by moxie0** [comment]
Closing this for now, but please reopen if you get a chance to incorporate @mcginty's feedback.

**Pull request closed without merging**